esp32-firmware/src/main.cpp

```
@@ -26,39 +35,130 @@ const int MIN_ZERO_CROSSINGS = 2;           // Min 2 przejścia = 1 pełny okres
+
+void adcSamplingTask(void * pvParameters) {
+    static uint8_t dmaBuffer[ADC_DMA_FRAME_BYTES];
+    static int frameU[SAMPLES];
+    static int frameI[SAMPLES];
+    int idxU = 0;
+    int idxI = 0;
+
+    for (;;) {
+        uint32_t bytesRead = 0;
+        esp_err_t err = adc_digi_read_bytes(dmaBuffer, sizeof(dmaBuffer), &bytesRead, 1000);
+        if (err == ESP_ERR_TIMEOUT) {
```

Clang (14.0.6)

[note] +2, including nesting penalty of 1, nesting level increased to 2

(clang)